Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -394,6 +394,10 @@ class BrowserViewController: UIViewController, BrowserViewControllerDelegate {
             let provider = self.braveCore.walletProvider(with: self, isPrivateBrowsing: tab.isPrivate) else {
         return nil
       }
+      if let keyringService = BraveWallet.KeyringServiceFactory.get(privateMode: tab.isPrivate) {
+        tab.walletKeyringService = keyringService
+        keyringService.add(tab)
+      }
```

Comment:
Maybe better to put this where `makeWalletProvider` is actually called, not inside the closure